lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_strings_file_validation_helper.rb

```diff
@@ -70,8 +70,8 @@ def self.find_duplicated_keys(file:)
 
           state = State.new(context: :root, buffer: StringIO.new, in_escaped_ctx: false, found_key: nil)
 
-          File.readlines(file).each_with_index do |line, line_no|
-            line.chars.each_with_index do |c, col_no|
+          File.readlines(file, mode: 'rb:BOM|UTF-8').each_with_index do |line, line_no|
+            line.encode('UTF-8').chars.each_with_index do |c, col_no|
```

Comment:
> I've already used this technique and notation (documented clearly in the IO.read/File.read/… ruby documentation) in other places of the codebase, and this basically allows to read the file using the text encoding it's saved as, instead of assuming and hardcoding ASCII or UTF-8.

I wonder if it'd be worth making a helper like `ios_strings_file_lines(path)` that wraps this approach? Not a blocker, just a thought 🙂

Reply:
Yeah, maybe 🤔 . Good thought

I only used this `rb:BOM|UTF-8` technique with `File.readlines` in two places — the other being in `ios_l10n_helper::merge_strings` method, where it's used similarly but slightly differently due to an `unless` condition. And the only other place I've used that magic `mode` was with `File.read` not `File.readlines`. So not sure how that would be mutializable even for just the two instances where I use it with `File.readlines`, but could keep that idea in the back of my head especially if we start needing that one with `File.readlines` in more occasions 🤔

Reply:
@jkmassel Addressed in e6a05751f6330af3ca3e13a9f85f1ce9580f18f1